src/Common/Auth/AuthHash.php

```diff
@@ -192,6 +192,10 @@ public static function passwordVerify(&$password, $hash)
         if (!empty(preg_match('/^\$6\$rounds=/', $hash))) {
             // Process SHA512HASH algo separately, since uses crypt
             $valid = hash_equals($hash, crypt($password, $hash));
+        } else if (preg_match('/^\$2a\$05\$/', $hash)) {
+            $salt = explode("$", $hash);
+
+            $valid = hash_equals($hash, crypt($password, $salt));
         } else {
             // Process algos supported by standard password_verify
```

Comment:
hi @dcvidigal , nice catch! Note it is a bit more complicated since $2a$05 hashes in OpenEMR 5.0 and later work fine in upgrade, but as you noted $2a$05 hashes in pre-OpenEMR 5.0 do not work. In order to avoid breaking hashes from OpenEMR 5.0 and later would instead drop your code in the else if above and place the following code in this else block:
```
            // Process algos supported by standard password_verify
            $valid = password_verify($password, $hash);

            if (!$valid) {
                // Ensure do not need to process legacy hash (pre 5.0.0), which will get converted to standard hash
                //  after a successful auth. The legacy hash has a malformed salt/hash combination which needs to
                //  be adjusted and then run via crypt.
                if (!empty(preg_match('/^\$2a\$05\$/', $hash)) && (substr($hash,28, 1) === '.')) {
                    $fixedSalt = substr($hash,0, 28) . "$";
                    if (strlen($fixedSalt) !== 29) {
                        $valid = false;
                    } else {
                        $valid = hash_equals($hash, crypt($password, $fixedSalt));
                    }
                }
            }
```